tests/test-customize-changesets-controller.php

```diff
@@ -68,16 +137,211 @@ public function test_get_item_schema() {
+	}
+
+	/**
+	 * Test getting changeset without having proper permissions.
+	 */
+	public function test_get_item_without_permissions() {
+		$manager = new WP_Customize_Manager();
+		$manager->save_changeset_post();
+
+		wp_set_current_user( self::$subscriber_id );
+		$request = new WP_REST_Request( 'GET', sprintf( '/customize/v1/changesets/%s', $manager->changeset_uuid() ) );
+		$response = $this->server->dispatch( $request );
+
+		$this->assertErrorResponse( 'rest_forbidden', $response, 403 );
+	}
+
+	/**
+	 * Test getting changeset with invalid UUID.
+	 */
+	public function test_get_item_with_invalid_uuid() {
+
+		$request = new WP_REST_Request( 'GET', sprintf( '/customize/v1/changesets/%s', REST_TESTS_IMPOSSIBLY_HIGH_NUMBER ) );
+		$response = $this->server->dispatch( $request );
+
+		$this->assertErrorResponse( 'rest_no_route', $response, 404 );
+	}
+
+	/**
+	 * Test getting changeset list with edit context with proper permissions.
+	 */
+	public function test_get_item_list_context_with_permission() {
+		wp_set_current_user( self::$admin_id );
+		$request = new WP_REST_Request( 'GET', '/customize/v1/changesets' );
+		$request->set_query_params( array(
+			'context' => 'edit',
+		) );
+		$response = $this->server->dispatch( $request );
+		$this->assertNotInstanceOf( 'WP_Error', $response );
+		$response = rest_ensure_response( $response );
+		$this->assertEquals( 200, $response->get_status() );
+	}
+
+	/**
+	 * Test getting changeset list with edit context without proper permissions.
+	 */
+	public function test_get_item_list_context_without_permission() {
+		wp_set_current_user( self::$subscriber_id );
+		$request = new WP_REST_Request( 'GET', '/customize/v1/changesets' );
+		$request->set_query_params( array(
+			'context' => 'edit',
+		) );
+		$response = $this->server->dispatch( $request );
+
+		$this->assertErrorResponse( 'rest_forbidden_context', $response, 403 );
+	}
+
+	/**
+	 * Test getting a changeset with edit context without proper permissions.
+	 */
+	public function test_get_item_context_without_permission() {
+		$manager = new WP_Customize_Manager();
+		$manager->save_changeset_post();
+
+		wp_set_current_user( self::$subscriber_id );
+		$request = new WP_REST_Request( 'GET', sprintf( '/customize/v1/changesets/%s', $manager->changeset_uuid() ) );
+		$request->set_query_params( array(
+			'context' => 'edit',
+		) );
+		$response = $this->server->dispatch( $request );
+
+		$this->assertErrorResponse( 'rest_forbidden_context', $response, 403 );
 	}
 
+	// @todo get items by author and other query params.
 	/**
 	 * Test get_items.
 	 *
 	 * @covers WP_REST_Customize_Changesets_Controller::get_items()
 	 */
 	public function test_get_items() {
-		$this->markTestIncomplete();
+		wp_set_current_user( self::$admin_id );
+
+		$manager1 = new WP_Customize_Manager();
+		$manager1->save_changeset_post();
+
+		$manager2 = new WP_Customize_Manager();
+		$manager2->save_changeset_post();
```

Comment:
FYI: You might want to consider doing `$this->factory()->post->create()` and create the changeset post “manually” since there is a problem with instantiating multiple `WP_Customize_Manager` instances in that it will add additional hooks which can cause problems. In this case here for the unit test it is fine, but @patelutkarsh and I ended up interfacing directly with the underlying `WP_Post` data in https://github.com/xwp/wordpress-develop/pull/226/commits/7077ef1154bce52a9c0f29c0e5732edbb84589fb

We eventually need to create a separate `WP_Customize_Changeset` class to handle modeling the post type.

You could add a method like `create_changeset` which then contains inside of it the `$this->factory()->post->create( array( 'post_type' => 'customize_changeset', 'post_name' => wp_generate_uuid4(), … ) )`.